### PR TITLE
Use MemoryBackingStore instead of MARF

### LINKED ISSUE #17: Database operations should use a MemoryBackingStore rather than a MARF
Database operations should not use an actual MARF backend in the benchmarks: these operations are _already_ charged for their MARF operations by being assessed a marf read or write in those cost dimensions. The _runtime_ component of their cost should not incorporate any MARF overhead.

These operations are:

    cost_at_block
    cost_create_ft
    cost_block_info
    cost_stx_balance
    cost_stx_transfer
    cost_ft_mint
    cost_ft_transfer
    cost_ft_balance
    cost_ft_get_supply
    cost_ft_burn
    poison_microblock
    cost_analysis_storage
    cost_analysis_use_trait_entry
    cost_analysis_get_function_entry
    cost_load_contract
    cost_create_map
    cost_create_var
    cost_create_nft
    cost_fetch_entry
    cost_set_entry
    cost_fetch_var
    cost_set_var
    cost_contract_storage
    cost_nft_mint
    cost_nft_transfer
    cost_nft_owner
    cost_nft_burn

### PR DESCRIPTION
**Pull request by reedrosenbluth**
fixes #17

### CODE REVIEW

**Review by kantai — APPROVED** [review]
These changes look good to me -- the performance of the data operations is still a little higher than I would have thought, but the numbers make sense at the moment.

File: benches/bench.rs [hunk 1]
```diff
@@ -2294,13 +2211,11 @@ fn bench_create_map(c: &mut Criterion) {
     let mut group = c.benchmark_group(function.to_string());
 
     for input_size in &INPUT_SIZES {
-        // want a warmed-up marf
```
Comment (pavitthrap) [inline benches/bench.rs:2297]: Do we preserve comments of this type (doesn't have to be exactly /`/ want a warmed-up marf`) - could be helpful to denote we are doing something "different" in some functions for a reason

**Review by pavitthrap — APPROVED** [review]
LGTM!

File: benches/bench.rs (hunk 1 above)
Comment (reedrosenbluth) [reply benches/bench.rs:2297]: we are really only ever doing 1 of 3 things now:

1. Using a fresh MemoryBackingStore
2. ^ + initializing a SimHeadersDB
3. Passing in a MemoryBackingStore with preloaded state

I think 1 and 2 are pretty self explanatory, and 3 only happens twice I think.

**Pull request merged**